Solutions/Okta Single Sign-On/Hunting Queries/LegacyAuthentication.yaml

```diff
@@ -0,0 +1,32 @@
+id: 6a9199ec-bc32-4935-9f82-4aa848edb3fc
+name: Okta login attempts using Legacy Auth
+description: |
+  'This query identifies use of legacy authentication protocol in the Okta Logs.'
+requiredDataConnectors:
+  - connectorId: OktaSSO
+    dataTypes:
+      - Okta_CL
+tactics:
+  - CredentialAccess
+relevantTechniques:
+  - T1556
+tags:
+  - Okta
+query: |
+ Okta_CL
+ | where debugContext_debugData_requestUri_s  has 'sso/wsfed/active'
+ | where outcome_result_s == 'SUCCESS'
+ | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated) by actor_alternateId_s, actor_displayName_s, client_userAgent_os_s, client_ipAddress_s, client_geographicalContext_state_s,displayMessage_s, outcome_result_s,eventType_s,
+ outcome_reason_s, column_ifexists('debugContext_debugData_logOnlySecurityData_s', ""), debugContext_debugData_threatSuspected_s, client_userAgent_rawUserAgent_s,client_userAgent_browser_s, severity_s,  client_geographicalContext_country_s, client_geographicalContext_city_s
+ | extend Account_0_Name = actor_displayName_s
+ | extend IP_0_Address = client_ipAddress_s
+entityMappings:
+  - entityType: Account
+    fieldMappings:
+      - identifier: Name
+        columnName: actor_displayName_s
```

Comment:
Is this a full UPN or just a username?

Reply:
When we think of UPN we normally think of username and domain combined with the @ separator. In the Okta logs I didn't see the identity for most events in the UPN format. So, I have added two identifiers for the Account entity. A **Name** which can be though of as the unique name of an account similar to our alias and a **DisplayName** that is the complete name of the account.